cmd/debugger/main.go

```diff
@@ -84,6 +84,45 @@ func populateShellHistory(cmd string) error {
 	return result
 }
 
+func sendFile(ctx context.Context, sockAddr, src, dst string) error {
+	log := slog.GetLogger(ctx)
+
+	conn, err := net.Dial("unix", sockAddr)
+	if err != nil {
+		return errors.Wrap(err, "failed to connect to remote debugger")
+	}
+	defer func() {
+		err := conn.Close()
+		if err != nil {
+			log.Error(errors.Wrap(err, "error closing"))
+		}
+	}()
+
+	// send a protocol version
+	err = common.WriteDataPacket(conn, 0x01, nil)
+	if err != nil {
+		return err
+	}
+
+	err = common.WriteDataPacket(conn, len(dst), []byte(dst))
+	if err != nil {
+		return err
+	}
+
+	b, err := os.ReadFile(src)
+	if err != nil {
+		return err
+	}
+
+	err = common.WriteDataPacket(conn, len(b), b)
+	if err != nil {
+		return err
+	}
+
+	// send end of file packet
+	return common.WriteDataPacket(conn, 0x00, nil)
+}
```

Comment:
Wondering if we should do anything special about backwards compatibility. If we're working against a satellite that is on an older version, the debugger will be of that older version. Maybe we don't have to do anything though, since this is experimental for now.

Reply:
so far we only have a single version:

```
	// send a protocol version
	err = common.WriteDataPacket(conn, 0x01, nil)
	if err != nil {
		return err
	}
```

Then once we expand the version, the client would populate a prioritised list of accepted protocols in `DebuggerSettings` (defined under https://github.com/earthly/earthly/blob/main/debugger/common/settings.go#L10 )